src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs

```diff
@@ -340,6 +346,69 @@ internal async void SetLabels()
                $"showScreenAgainLabel: '{Wpf.Properties.Resources.SplashScreenShowScreenAgainLabel}'" + "})");
         }
 
+        /// <summary>
+        /// At Dynamo startup process load the preferences settings file located in C:\ProgramData\Dynamo
+        /// </summary>
+        internal void LoadPreferencesFileAtStartup()
+        {
+            if (viewModel.PreferenceSettings.IsFirstRun == true)
+            {
+                //Move the current location two levels up
+                var programDataDir = Directory.GetParent(Directory.GetParent(viewModel.Model.PathManager.CommonDataDirectory).ToString()).ToString();
```

Comment:
Would you make it a property not that it should be but that way I imagine we can add unit test for it but not direct splash screen to import setting from a different test location? Anyway, I would like a unit test to guard this behavior if possible.

Reply:
@QilongTang 
Then my understanding is that you are requesting a unit test which validates that the NewProperty added:

NewProperty = Directory.GetParent(Directory.GetParent(viewModel.Model.PathManager.CommonDataDirectory).ToString()).ToString();

is always pointing to the path "C:\ProgramData\Dynamo".
is that correct?

Reply:
@RobertGlobant20 Sorry for the confusion, I would like a unit test to valid if initializing Splash Screen with isFirstTime = true, we will try to import a setting for a defined property path, does it make sense?

Reply:
@QilongTang 
I just did a quick analysis of creating an UI Unit Test for SplashScreen and the problem that I can see is that DynamoTestUIBase class (base class for most of the UI unit tests) is always creating the DynamoView and DynamoModel instances so we need to create a new base UI test class (something like SplashScreenDynamoUIBase class) or modify the current one (that could affect all the UI Unit Tests that derive from DynamoTestUIBase ) to create the SplashScreen instance and programatically click the "Launch Dynamo" button (in the SplashScreen).
Also another thing is that we need to copy/create a xml file to C:\ProgramData\Dynamo (not sure if we have permissions to do that in the CI server) and that needs to happen before the SplashScreen OnStaticScreenReady event handler is executed (we can do that in the new SplashScreenDynamoUIBase class before creating the SplashScreen instance).
Then I think we should consider create a new Jira task for coding the necessary infrastructure for creating SplashScreen tests and analyze all the possible solutions (so we don't affect existing UI Tests).
Please let me know your thoughts about it. Thanks
FYI:  @filipeotero @jesusalvino @reddyashish @zeusongit @avidit

Reply:
@RobertGlobant20 I think the new UITestBase class is more of an issue than the file path. The second issue you mentioned is exactly why I recommend you moving the path to a property so we can ulter the default behavior when running tests.
I agree on creating a new Jira task. Let's catch up on Monday and I will merge this for now.